src/components/ComponentSettingsModal.jsx

```diff
@@ -0,0 +1,467 @@
+import React, { Component } from 'react';
+import axios from 'axios';
+import { Base64 } from 'js-base64';
+import PropTypes from 'prop-types';
+import * as _ from 'lodash';
+import { Redirect } from 'react-router-dom';
+import FormField from './FormField';
+import FormFieldFile from './FormFieldFile';
+import LoadingButton from './LoadingButton';
+import {
+  frontMatterParser,
+  concatFrontMatterMdBody,
+  enquoteString,
+  dequoteString,
+  generateResourceFileName,
+  generateCollectionPageFileName,
+  generatePageFileName,
+  retrieveCollectionAndLinkFromPermalink,
+} from '../utils';
+import elementStyles from '../styles/isomer-cms/Elements.module.scss';
+import { validatePageSettings, validateResourceSettings } from '../utils/validators';
+import DeleteWarningModal from './DeleteWarningModal';
+
+// axios settings
+axios.defaults.withCredentials = true
+
+export default class ComponentSettingsModal extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      title: '',
+      permalink: '',
+      fileUrl: null,
+      date: '',
+      mdBody: null,
+      category: '',
+      sha: '',
+      allCategories: null,
+      type: '',
+      errors: {
+        title: '',
+        permalink: '',
+        fileUrl: '',
+        date: '',
+        category: '',
+        prevCategory: '',
+      },
+      canShowDeleteWarningModal: false,
+      baseApiUrl: '',
+      prevCategory: '',
+      thirdNavTitle: '',
+      redirectToNewPage: false,
+      newPageUrl: '',
+      isPost: true,
+    };
+  }
+
+  async componentDidMount() {
```

Comment:
the `componentDidMount` section has gotten a little long - can we have a few comments to guide the reader on the component mount flow

Reply:
Resolved in https://github.com/isomerpages/isomercms-frontend/pull/184/commits/68695d56681bd2850144fa65576b6b72a18d4538